api/coreservice.go

```diff
@@ -1340,8 +1340,12 @@ func (core *coreService) LogsInBlockByHash(filter *logfilter.LogFilter, blockHas
 }
 
 // LogsInBlock filter logs in the block x
-func (core *coreService) LogsInBlock(filter *logfilter.LogFilter, blockNumber uint64) ([]*iotextypes.Log, error) {
-	return core.logsInBlock(filter, blockNumber)
+func (core *coreService) LogsInBlock(filter *logfilter.LogFilter, blockNumber uint64, c chan []*iotextypes.Log) {
```

Comment:
It is unnecessary to modify Public method `LogsInBlock` for this issue. `logsInBlock()` could be directly used in `LogsInRange()` if you want

Reply:
I'll use this approach

Reply:
> How about put `LogInBlock` in a goroutine function literal and try not to modify the original code structure

you could have a try